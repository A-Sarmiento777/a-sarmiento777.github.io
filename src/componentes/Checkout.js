import React from 'react';

const Checkout = ({ carrito, removeItemFromCart, clearCart }) => {

    let amount = carrito?.map(a => a.price)
    let totalAmount = amount?.reduce((a, b) => a + b, 0)

    return (
        <div>
            <h3>Carrito de Compras</h3>
            {carrito.length ? <table style={{ width: '100%' }}>
                <thead>
                    <tr>
                        <td style={{ fontWeight: 'bold' }}>Product</td>
                        <td style={{ fontWeight: 'bold' }}>Title</td>
                        <td style={{ fontWeight: 'bold' }}>Quantity</td>
                        <td style={{ fontWeight: 'bold' }}>Price</td>
                        <td style={{ fontWeight: 'bold' }}>Remove</td>
                    </tr>
                </thead>
                <tbody>
                    {carrito?.map(producto => {
                        return (
                            <>
                                <tr>
                                    <td><img src={process.env.PUBLIC_URL + `/Assets/${producto.type}/${producto.nombre}.jpg`} width='150' /></td>
                                    <td>{producto.nombre}</td>
                                    <td>{producto.cantidad}</td>
                                    <td>{producto.price}</td>
                                    <td onClick={() => removeItemFromCart(producto.id)} style={{ color: 'red' }}>Remove</td>
                                </tr>
                            </>
                        )
                    })}
                </tbody>
            </table> : <h4>Nothing in cart! Please Add some items</h4>}

            {carrito.length ? <div style={{ display: 'flex', flexDirection: 'column', alignItems: 'flex-end' }}>
                <div>
                    Total Amount: {totalAmount}
                </div>
                <div>
                    After Tax: {totalAmount - ((totalAmount * 30) / 100)}
                </div>
                <div>
                    <button style={{ backgroundColor: 'black', color: 'white' }} onClick={() => {
                        alert('Thanks for the Shopping')
                        clearCart()
                    }}>Checkout</button>
                </div>
            </div> : null}

        </div >
    );
}


export default Checkout;