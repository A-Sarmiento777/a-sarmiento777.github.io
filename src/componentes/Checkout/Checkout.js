import React, { useEffect } from 'react';
import { useHistory } from 'react-router';
import CheckoutCard from '../CheckoutCard/CheckoutCard';

const Checkout = ({priceFormatter, removeItemFromCart, increment, decrement, clearCart, getCart, cart }) => {

    const history = useHistory()
	let amount = cart && cart?.length && cart.map(a => parseInt(a.price) * parseInt(a.quantity))
	let totalAmount = amount && amount.length && amount?.reduce((a, b) => a + b, 0)
	useEffect(() => {
		getCart()
        // eslint-disable-next-line react-hooks/exhaustive-deps
	}, [])
    return (
        <div className="card p-4 mt-5 pt-4 mb-5 rounded" style={{minWidth:"460px"}}>
            {cart.length ? <table style={{ width: '100%' }}>
                <thead>
                    <tr>
                        <td style={{ fontWeight: 'bold' }}>Product</td>
                        <td style={{ fontWeight: 'bold' }}>Title</td>
                        <td style={{ fontWeight: 'bold' }}>Quantity</td>
                        <td style={{ fontWeight: 'bold' }}>Price</td>
                        <td style={{ fontWeight: 'bold' }}>Remove</td>
                    </tr>
                </thead>
                <tbody>
                    {cart?.map( (producto, index) => {
                        return (
                            <CheckoutCard priceFormatter={priceFormatter} removeItemFromCart={removeItemFromCart} increment={increment} decrement={decrement} getCart={getCart} cart={cart} producto={producto}/>
                        )
                    })}
                </tbody>
            </table> : <h4>Nothing in cart! Please Add some items</h4>}

            {cart.length ? 
                <div  className="mx-auto">
                <div>
                Price: {priceFormatter(totalAmount - ((totalAmount * 25) / 100))}
                    
                </div>
                <div>
                Incl tax: {priceFormatter(totalAmount)}

                </div>
                <div className="text-center mx-auto">
                    <button className="p-1 rounded" style={{ backgroundColor: 'black', color: 'white' }} onClick={() => {
                        history.push('/order')
                        clearCart()
                    }}>Checkout</button>
                </div>
                </div>
            : null}

        </div >
    );
}


export default Checkout;