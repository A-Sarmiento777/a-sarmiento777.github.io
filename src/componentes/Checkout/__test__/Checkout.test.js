import React from "react";
import { act, create, waitFor } from "react-test-renderer";
import Checkout from "../Checkout";

test("It should render completed Checkout", async () => {
  jest.spyOn(global, "fetch").mockImplementation(() => {
    return Promise.resolve({
      json: () => Promise.resolve([]),
    });
  });

  let testRenderer;
  await act(async () => {
    testRenderer = create(
        <Checkout priceFormatter={() => {}} removeItemFromCart={() => {}} increment={() => {}} decrement={() => {}} removeItemFromCartFav={() => {}} clearCart={() => {}} getCart={() => {}} cart={[]} />
        );
  });

  let tree = testRenderer.toJSON();
  expect(tree).toMatchSnapshot();
//   expect(global.fetch).toHaveBeenCalledTimes(1);
});
