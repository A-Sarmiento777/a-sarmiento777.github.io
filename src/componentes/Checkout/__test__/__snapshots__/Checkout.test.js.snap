// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render completed Checkout 1`] = `
<div
  className="card p-4 mt-5 pt-4 mb-5 rounded"
  style={
    Object {
      "minWidth": "460px",
    }
  }
>
  <h4>
    Nothing in cart! Please Add some items
  </h4>
</div>
`;
